### ISSUE TITLE
Concat label embed RAM++

### ISSUE THREAD

**Issue by pribadihcr** [xinyu1205/recognize-anything#154]
Hi, can we concat the label_embed between the default ram tag and open set / custom tag? thanks

**Comment by xinyu1205** [comment]
Yes, you can select or add the label_embed as you wish, which will not compromise its performance.

**Comment by pribadihcr** [comment]
question, why in this line it did normalization https://github.com/xinyu1205/recognize-anything/blob/main/ram/utils/openset_utils.py#L357 ?, It mean one tag have one label embeding,  while in default inference look like there are 233835 (4585*51) label embedding. Thanks

**Comment by pribadihcr** [comment]
another question, does the tag's accuracy will be depend on the descriptions?

**Comment by jurijsnazarovsambientai** [comment]
Yes, I have the same question. With open_set function we normalize embeddings of all descriptions. So, if we have 3 new tags, then embedding will go from 3*51 (given we use 51 description) to 3. While the original model uses (4585*51) label embedding. Is that a bug or done on purpose?

**Comment by jurijsnazarovsambientai** [comment]
@pribadihcr I actually believe that combining would not work, because of this line: https://github.com/xinyu1205/recognize-anything/blob/8bda8346ac764fbd91cca6a8a0a43a6fef40e73f/ram/models/ram_plus.py#L217 . There we rely on number of descriptions per tag, and if we append normalized tags from open_set, it will give us a wrong des_per_class. Thus, will continue with wrong computation until here: https://github.com/xinyu1205/recognize-anything/blob/8bda8346ac764fbd91cca6a8a0a43a6fef40e73f/ram/models/ram_plus.py#L225 . 

Seems like the only solution is to generate exactly 10 samples per tag in the open set and remove normalization, so that each tag has 51 descriptions. Then we can combine. 

What do you think, @pribadihcr @xinyu1205 ?
